NOnion/Network/TorCircuit.fs

```diff
@@ -92,14 +98,20 @@ type TorCircuit
                 let onionList, destination =
                     match customDestinationOpt with
                     | None ->
-                        let reversedNodesList = nodesStates |> List.rev
-                        reversedNodesList, reversedNodesList |> List.head
+                        let reversedNodesList = nodesStates |> Seq.rev
+
+                        let destinationNodeOpt =
+                            reversedNodesList |> Seq.tryHead
 
+                        match destinationNodeOpt with
+                        | None -> failwith "BUG: circuit has no nodes"
+                        | Some destinationNode ->
+                            reversedNodesList, destinationNode
                     | Some destination ->
                         nodesStates
-                        |> List.takeWhile(fun node -> node <> destination)
-                        |> List.append(List.singleton destination)
-                        |> List.rev,
+                        |> Seq.takeWhile(fun node -> node <> destination)
+                        |> Seq.append(Seq.singleton destination)
+                        |> Seq.rev,
```

Comment:
there's no need to change this to Seq; you can still use Seq's functions on a List

Reply:
ah, fair enough